app/src/main/java/com/rockthevote/grommet/ui/registration/RegistrationActivity.java

```diff
@@ -173,21 +173,32 @@ private void showCancelDialog() {
     public void onPreviousClick(View v) {
         int curPage = viewPager.getCurrentItem();
         if (curPage > 0) {
-            viewPager.setCurrentItem(curPage - 1);
+            int previousItem = curPage - 1;
```

Comment:
I should have handled all of this w/in the fragments, just notify them when they're about to be displayed or about to not be displayed, and either validate/load accordingly, it would be much more deterministic.

Reply:
I can see that being an advantage as the use case grows. I think it's still reasonably small enough to handle here, but would be a good improvement going forward.